src/gui/src/globalConnectDialog.cpp

```diff
@@ -0,0 +1,305 @@
+};
+
+static int toValue(GlobalConnectField f)
+{
+  return static_cast<int>(f);
+}
+
+/////////
+
+GlobalConnectDialog::GlobalConnectDialog(odb::dbBlock* block, QWidget* parent)
+    : QDialog(parent),
+      block_(block),
+      layout_(new QGridLayout),
+      add_(new QPushButton(this)),
+      clear_(new QPushButton(this)),
+      run_(new QPushButton(this)),
+      rules_({}),
+      inst_pattern_(new QLineEdit(this)),
+      pin_pattern_(new QLineEdit(this)),
+      net_(new QComboBox(this)),
+      region_(new QComboBox(this))
+{
+  setWindowTitle("Global Connect Rules");
+
+  QVBoxLayout* layout = new QVBoxLayout;
+  layout->addLayout(layout_);
+  QHBoxLayout* button_layout = new QHBoxLayout;
+  button_layout->addWidget(clear_);
+  button_layout->addWidget(run_);
+  layout->addLayout(button_layout);
+
+  setLayout(layout);
+
+  add_->setIcon(QIcon(":/add.png"));
+  add_->setToolTip(tr("Run"));
+  add_->setAutoDefault(false);
+  add_->setDefault(false);
+
+  run_->setIcon(QIcon(":/play.png"));
+  run_->setToolTip(tr("Run"));
+  run_->setAutoDefault(false);
+  run_->setDefault(false);
+
+  clear_->setIcon(QIcon(":/delete.png"));
+  clear_->setToolTip(tr("Clear"));
+  clear_->setAutoDefault(false);
+  clear_->setDefault(false);
+
+  connect(add_, SIGNAL(pressed()), this, SLOT(makeRule()));
+  connect(run_, SIGNAL(pressed()), this, SLOT(runRules()));
+  connect(clear_, SIGNAL(pressed()), this, SLOT(clearRules()));
+
+  layout_->addWidget(new QLabel("Instance pattern", this),
+                     0,
+                     toValue(GlobalConnectField::Instance),
+                     Qt::AlignCenter);
+  layout_->addWidget(new QLabel("Pin pattern", this),
+                     0,
+                     toValue(GlobalConnectField::Pin),
+                     Qt::AlignCenter);
+  layout_->addWidget(new QLabel("Net", this),
+                     0,
+                     toValue(GlobalConnectField::Net),
+                     Qt::AlignCenter);
+  layout_->addWidget(new QLabel("Region", this),
+                     0,
+                     toValue(GlobalConnectField::Region),
+                     Qt::AlignCenter);
+  for (auto* rule : block_->getGlobalConnects()) {
+    addRule(rule);
+  }
+
+  region_->addItem("All",
+                   QVariant::fromValue(static_cast<odb::dbRegion*>(nullptr)));
+  for (auto* region : block_->getRegions()) {
+    region_->addItem(QString::fromStdString(region->getName()),
+                     QVariant::fromValue(static_cast<odb::dbRegion*>(region)));
+  }
+
+  for (auto* net : block_->getNets()) {
+    if (!net->isSpecial()) {
+      continue;
+    }
+    net_->addItem(QString::fromStdString(net->getName()),
+                  QVariant::fromValue(static_cast<odb::dbNet*>(net)));
+  }
+  for (auto* net : block_->getNets()) {
+    if (net->isSpecial()) {
+      continue;
+    }
+    net_->addItem(QString::fromStdString(net->getName()),
+                  QVariant::fromValue(static_cast<odb::dbNet*>(net)));
+  }
```

Comment:
Should this be skip if _not_ special?

Reply:
In the previous block, it's adding all the special nets first. This is just to fill in the remainder.
Technically, the global connections do not have to be special nets, but I can limit the list in the GUI to just those as you asked? Does that make sense to you?

Reply:
The list is too long to be useful.  If we want to support all nets then some sort of searching mechanism is needed.  I am ok with just limiting it to special nets as I think that is the only real use case at this point that I know of.

Reply:
I limited the list to just special nets